src/CLIFramework/Extension/ExtensionBase.php

```diff
@@ -0,0 +1,42 @@
+<?php
+namespace CLIFramework\Extension;
+
+use CLIFramework\ServiceContainer;
+use CLIFramework\Command;
+use CLIFramework\Hook\Hookable;
+use CLIFramework\Hook\HookHolder;
+use CLIFramework\Exception\ExtensionException;
+
+abstract class ExtensionBase
+    implements Extension, Hookable
+{
+    private $hooks;
+
+    private function getHooks()
```

Comment:
I think we can bump our requirement to 5.4 since 5.3 is too old right now. :)

the synopsis  above works for me :)